From 1b29a12f0393fc02bb3560c8fd9cf3a475a0db5f Mon Sep 17 00:00:00 2001
From: Afif Elghraoui <afif@debian.org>
Date: Sat, 16 Jul 2016 14:09:10 -0700
Subject: [PATCH] debcommit: use standard version mangling for git tags

Closes: #770187

This patch implements the changes requested in #770187.
Please see the bug report for details about the rationale.
---
 scripts/debcommit.pl | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/scripts/debcommit.pl b/scripts/debcommit.pl
index ce22100..2a32576 100755
--- a/scripts/debcommit.pl
+++ b/scripts/debcommit.pl
@@ -700,8 +700,8 @@ sub tag {
 	}
     }
     elsif ($prog eq 'git') {
-	$tag =~ s/^[0-9]+://; # strip epoch
-	$tag =~ tr/~/./; # mangle for git
+	$tag =~ tr/~/_/; # mangle for git
+	$tag =~ tr/:/%/;
 	if ($tag =~ /-/) {
 	    # not a native package, so tag as a debian release
 	    $tag = "debian/$tag";
-- 
2.1.4


